### PR TITLE
Add `skip_while_next` lint

### LINKED ISSUE #4036: Improvement Idea: the ilter_next should also cover skip_while().next()
I recently realized that `skip_while(|i| !condition(i)).next()` does the same as `filter(|i| condition(i)).next()` and more importantly has the same intent. 

I propose to also lint `skip_while(...).next()`with the filter_next lint.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #4036

changelog: Add `skip_while_next` lint

### CODE REVIEW

File: clippy_lints/src/methods/mod.rs [hunk 1]
```diff
@@ -2530,6 +2555,20 @@ fn lint_filter_next<'a, 'tcx>(
...
+fn lint_skip_while_next<'a, 'tcx>(
+    cx: &LateContext<'a, 'tcx>,
+    expr: &'tcx hir::Expr<'_>,
+    _skip_while_args: &'tcx [hir::Expr<'_>],
+) {
+    // lint if caller of `.skip_while().next()` is an Iterator
+    if match_trait_method(cx, expr, &paths::ITERATOR) {
+        let msg = "called `skip_while(p).next()` on an `Iterator`. \
+                   This is more succinctly expressed by calling `.find(!p)` instead.";
+        span_lint(cx, SKIP_WHILE_NEXT, expr.span, msg);
```
Comment (flip1995) [inline clippy_lints/src/methods/mod.rs]: Can you split up the message and use `span_help_and_lint`?

Comment (JohnTitor) [reply clippy_lints/src/methods/mod.rs]: Yeah, it's reasonable, addressed. And I've realized that some similar lints also use `span_lint`. I'll follow up later.

Comment (flip1995) [reply clippy_lints/src/methods/mod.rs]: Do you want to do this in this PR or open a new one?

**Comment by flip1995** [comment]
@bors delegate+

r=me if you want to split up the messages of other lints in a separate PR, instead of in this PR.

**Comment by JohnTitor** [comment]
Ah sorry, "later" was a little vague. I intended to follow up on another PR. So let's
@bors r=flip1995

**Pull request merged**